src/UI/Implementation/Component/Input/DynamicInputDataIterator.php

```diff
@@ -0,0 +1,75 @@
+<?php declare(strict_types=1);
+
+/* Copyright (c) 2022 Thibeau Fuhrer <thibeau@sr.solutions> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\UI\Implementation\Component\Input;
+
+use Iterator;
+
+/**
+ * @author Thibeau Fuhrer <thibeau@sr.solutions>
+ */
+class DynamicInputDataIterator implements Iterator
+{
+    protected string $parent_input_name;
+    protected array $post_data;
+    protected int $index = 0;
+
+    public function __construct(InputData $data, string $parent_input_name)
+    {
+        $this->post_data = $data->getOr($parent_input_name, []);
+        $this->parent_input_name = $parent_input_name;
+    }
+
+    public function current() : ?InputData
+    {
+        if ($this->valid()) {
+            $entry = [];
+            $name_source = new DynamicInputsNameSource($this->parent_input_name);
```

Comment:
`Group::withNameFrom()` calls the parent method which set's the groups name, though it is never used except I am missing something?. The problem would resolve if we just change this method not to call the parent's method.

Reply:
All tests pass even if the Group input does never have a name. However I had do adjust the Optional- and Switchable Group to use `Input::withNameFrom()` because it seems they do need it and extended it from Group. But other than this I think the issue should be resolved.